### PR TITLE
 time ticker instand  sleep

### PR DESCRIPTION
**Pull request by jasonchuan**
time sleep is a  waste of cpu time ,so take time ticker  instand

### CODE REVIEW

**Review by chuntaojun — APPROVED** [review]

File: clients/naming_client/service_info_updater.go [hunk 1]
```diff
@@ -44,23 +44,26 @@ func NewServiceInfoUpdater(serviceInfoHolder *naming_cache.ServiceInfoHolder, up
 
 func (s *ServiceInfoUpdater) asyncUpdateService() {
 	sema := util.NewSemaphore(s.updateThreadNum)
+	ticker := time.NewTicker(time.Second)
 	for {
```
Comment (chuntaojun) [inline clients/naming_client/service_info_updater.go:48]: add defer ticker.Stop()

**Review by binbin0325 — COMMENTED** [review]
sleep & ticker bottom layer is timer, no difference.
how is ticker better than sleep?

**Comment by jasonchuan** [comment]
> sleep & ticker bottom layer is timer, no difference. how is ticker better than sleep?
you are right , sleep func uses timer cache in g  in go 1.9 version ,  sleep would new timer instance every time called before 1.9 version . Nacos sdk need after 1.15,  so is OK.Thank You ！   
the only one difference is that  using ticker will put timer instance on stack after escape analysis  ,sleep on heap ,  ticker is little  better   than sleep for GC

**Comment by binbin0325** [comment]
> 

So I don't see any improvement in pr at the moment, can I turn it off?

**Comment by binbin0325** [comment]
> > sleep & ticker bottom layer is timer, no difference. how is ticker better than sleep?
> > you are right , sleep func uses timer cache in g  in go 1.9 version ,  sleep would new timer instance every time called before 1.9 version . Nacos sdk need after 1.15,  so is OK.Thank You ！
> > the only one difference is that  using ticker will put timer instance on stack after escape analysis  ,sleep on heap ,  ticker is little  better   than sleep for GC

so I don't see any improvement in pr at the moment, can I turn it off?

**Comment by jasonchuan** [comment]
> 
OK

**Comment by binbin0325** [comment]
> > 
> 
> OK

**Pull request closed without merging**